core/controlplane/config/config.go

```diff
@@ -435,6 +438,7 @@ type DeploymentSettings struct {
 	CloudWatchLogging       `yaml:"cloudWatchLogging,omitempty"`
 	AmazonSsmAgent          `yaml:"amazonSsmAgent,omitempty"`
 	CloudFormationStreaming bool `yaml:"cloudFormationStreaming,omitempty"`
+	KubeDnsMasq             `yaml:"kubeDnsMasq,omitempty"`
```

Comment:
I'm ok with the depth if it provides long term api stability & extensibility but anyways,

Should all dnsmasq configured equally?(I just didn't know! 

Then, `kubeDns.nodeLocalResolverEnabled` would be good from my perspective.
If it doesn't look ok to you, I'm ok with `kubeDns.nodeLocalResolver` as a bool setting, as you've originally suggested!